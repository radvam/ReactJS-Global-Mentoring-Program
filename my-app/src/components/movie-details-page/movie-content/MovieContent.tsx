import React, { FC, useMemo } from "react";

import { Spin } from "antd";
import { Spinner } from "../../common/Spinner/Spinner.style";

import { MovieContentProps } from "./MovieContent.interface";

import {
  MovieContentWrapper,
  PosterBlock,
  PosterImage,
  DescriptionBlock,
  MovieHeader,
  MovieTitle,
  MovieRating,
  MovieTagLine,
  MovieDateAndDuration,
  MovieRealiseDate,
  MovieDuration,
  MovieDescription,
} from "./MovieContent.style";

export const MovieContent: FC<MovieContentProps> = ({
  movie,
  movieLoading,
}): React.ReactElement => {
  const poster_path = movie?.poster_path;
  const title = movie?.title;
  const vote_average = movie?.vote_average;
  const tagline = movie?.tagline;
  const overview = movie?.overview;
  const release_date = movie?.release_date;
  const runtime = movie?.runtime;

  const movieRealiseDate = useMemo(() => release_date?.slice(0, 4), [
    release_date,
  ]);

  return (
    <Spin spinning={movieLoading} indicator={<Spinner />}>
      <MovieContentWrapper>
        <PosterBlock>
          <PosterImage src={poster_path} alt="poster" />
        </PosterBlock>
        <DescriptionBlock>
          <MovieHeader>
            <MovieTitle>{title}</MovieTitle>
            <MovieRating>{vote_average}</MovieRating>
          </MovieHeader>
          <MovieTagLine>{tagline}</MovieTagLine>
          <MovieDateAndDuration>
            <MovieRealiseDate>{movieRealiseDate}</MovieRealiseDate>
            {runtime && <MovieDuration>{`${runtime} min`}</MovieDuration>}
          </MovieDateAndDuration>
          <MovieDescription>{overview}</MovieDescription>
        </DescriptionBlock>
      </MovieContentWrapper>
    </Spin>
  );
};
