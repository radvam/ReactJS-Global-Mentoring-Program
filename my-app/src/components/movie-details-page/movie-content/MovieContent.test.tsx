import * as React from "react";
import { render, screen } from "@testing-library/react";

import { MovieContent } from "./MovieContent";

const movie = {
  id: 338970,
  title: "Tomb Raider",
  tagline: "Her legend begins",
  vote_average: 6.2,
  vote_count: 817,
  release_date: "2018-03-08",
  poster_path:
    "https://image.tmdb.org/t/p/w500/ePyN2nX9t8SOl70eRW47Q29zUFO.jpg",
  overview:
    "Lara Croft, the fiercely independent daughter of a missing adventurer, must push herself beyond her limits when she finds herself on the island where her father disappeared.",
  budget: 94000000,
  revenue: 126025000,
  genres: ["Action", "Adventure"],
  runtime: 118,
};

const movieLoading = false;

describe("MovieContent", () => {
  test("should render", () => {
    const { asFragment } = render(
      <MovieContent movie={movie} movieLoading={movieLoading} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("title should be render", () => {
    render(<MovieContent movie={movie} movieLoading={movieLoading} />);

    expect(screen.queryByText("Tomb Raider")).toBeInTheDocument();
  });

  test("poster should be render", () => {
    render(<MovieContent movie={movie} movieLoading={movieLoading} />);

    expect(screen.getByRole("img").getAttribute("src")).toEqual(
      "https://image.tmdb.org/t/p/w500/ePyN2nX9t8SOl70eRW47Q29zUFO.jpg"
    );
  });

  test("runtime should be equal 90 min", () => {
    const testMovie = { ...movie, runtime: 0 };

    render(<MovieContent movie={testMovie} movieLoading={movieLoading} />);

    expect(screen.getByText("90 min")).toBeInTheDocument();
  });
});
