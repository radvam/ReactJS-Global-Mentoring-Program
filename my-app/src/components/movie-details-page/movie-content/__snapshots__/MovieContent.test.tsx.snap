// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieContent should render 1`] = `
<DocumentFragment>
  <div
    class="ant-spin-nested-loading"
  >
    <div
      class="ant-spin-container"
    >
      <div
        class="sc-gsTCUz kxBEbM"
      >
        <div
          class="sc-dlfnbm kpgvUO"
        >
          <img
            alt="poster"
            class="sc-hKgILt lnGJJi"
            src="https://image.tmdb.org/t/p/w500/ePyN2nX9t8SOl70eRW47Q29zUFO.jpg"
          />
        </div>
        <div
          class="sc-eCssSg fKOyAQ"
        >
          <div
            class="sc-jSgupP zOCup"
          >
            <p
              class="sc-gKsewC eTlXqI"
            >
              Tomb Raider
            </p>
            <div
              class="sc-iBPRYJ hXeVrs"
            >
              6.2
            </div>
          </div>
          <p
            class="sc-fubCfw gHjKuM"
          >
            Her legend begins
          </p>
          <div
            class="sc-pFZIQ dyicPg"
          >
            <p
              class="sc-jrAGrp"
            >
              2018
            </p>
            <p
              class="sc-kEjbxe"
            >
              118 min
            </p>
          </div>
          <p
            class="sc-fubCfw sc-iqHYGH gHjKuM dxLnJ"
          >
            Lara Croft, the fiercely independent daughter of a missing adventurer, must push herself beyond her limits when she finds herself on the island where her father disappeared.
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
